goldmane/pkg/aggregator/index.go

```diff
@@ -60,42 +61,97 @@ type IndexFindOpts struct {
 	filter *proto.Filter
 }
 
-func (idx *index[E]) List(opts IndexFindOpts) []*types.Flow {
+// List returns a list of flows and metadata about the list that's returned.
+// TODO we might want to return a channel instead of a list and make this more memory efficient. The list metadata
```

Comment:
TODOs no longer apply?

Reply:
Ya I wasn't suggesting keeping the today (I've removed it), just saying that it might apply for a future use case.